.github/workflows/flowzone.yml

```diff
@@ -387,6 +389,111 @@ jobs:
             echo "enabled=false" >> $GITHUB_OUTPUT
           fi
 
+      - name: Check for Python Poetry pyproject.toml
+        id: python_poetry
+        run: |
+          if test -f "pyproject.toml"
+          then
+            echo "found pyproject.toml"
+            if grep 'build-backend.*poetry' pyproject.toml
+            then 
+              echo "Poetry used"
+              echo "enabled=true" >> $GITHUB_OUTPUT
+              echo "PYTHON_VERSIONS=[]" >> $GITHUB_ENV
+            else
+              echo "Poetry not used"
+              echo "enabled=false" >> $GITHUB_OUTPUT
+            fi
+          else
+            echo "enabled=false" >> $GITHUB_OUTPUT
+          fi
+
+      # Check which Python 3.7+ versions meet the Poetry project requirements
```

Comment:
Would be awesome to have a loop here instead of hardcoding the versions. But I suppose that is not really feasible in this situation?

Reply:
I think looping is done via matrices and maybe there is a way but I followed the Node checks template to keep everything consistent.

Reply:
A matrix happens for the entire job @pranasziaukas / @gantonayde so it doesn't work so well for a job like this where we only want to matrix specific steps (if I'm wrong please tell me/ping on zulip because it'd be awesome to have per-step matrices!)

Reply:
Alas I could not make it work, as @Page- explained and as per their docs you cannot generate steps via loops at the moment, which is a bit sad.